### DISCUSSION TITLE
lessThan/greaterThan with multiple fields

### DISCUSSION THREAD

**Discussion by derolf** [isar/isar#81] [Help]
Consider the following class:

```Dart
@Collection()
class Person {
  int? id;

  @Index(composite: [CompositeIndex('firstName')])
  late String lastName;

  @Index()
  late String firstName;
}
```

The composite index induces a hierarchical lexical order like in a phone book on `lastName`, `firstName`.

How can I pick the next `N` person that follow a certain `lastName`, `firstName` in that order?

* Given `lastName`, `firstName`
* Select all person `p` with: `p.lastName == lastName && p.firstName > lastName || p.lastName > lastName`
* Sort by `p.lastName, p.firstName`
* Limit to `N`

**Comment by simc** [comment]
@derolf I'm currently working on generating the required extensions methods. You will be able to do the following:
```dart
isar.persons.where()
  .lastNameEqualToFirstNameGreaterThan(lastName, lastName)
  .or()
  .lastNameGreaterThan(lastName)
  .limit(N)
  .findAll()
```

Isar Core already supports that. You can already test it with dynamic queries (`isar.persons.buildQuery()`)
